maas/resource_maas_user.go

```diff
@@ -125,3 +167,22 @@ func getUser(client *client.Client, userName string) (*entity.User, error) {
 
 	return nil, fmt.Errorf("user (%s) was not found", userName)
 }
+
+func verifyUserValid(client *client.Client, d *schema.ResourceData) error {
+	// ensure the user is a valid target for create/delete
+	me, err := client.Users.Whoami()
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	if name == me.UserName {
+		return fmt.Errorf("cannot operate on the currently logged in user %q", me.UserName)
+	}
+
+	if !me.IsLocal {
```

Comment:
The check here is wrong. Thinking about it more, the check should only happen at import. Reasoning: if Terraform tries to create a non local user or the same user as the owner of the api_key MAAS will throw an error either way. In both cases, a user with that username or email already exists. So there is no need to check during create.

For import the check is a different one than what this function is doing:

We have to confirm that there is a user in MAAS with the `d.Id()`, and then check that the user is not me (whoami) or a non local user. Pseudocode

```go
user, err := getUser(client, d.Id())
me, err := client.Users.Whoami()

if user.IsLocal {error}
if user.UserName == me.UserName {error}
```